GeneralsMD/Code/GameEngineDevice/Source/StdDevice/Common/StdLocalFileSystem.cpp

```diff
@@ -0,0 +1,342 @@
+//	if (!(access&File::WRITE)) {
+//		// Return a ramfile.
+//		RAMFile *ramFile = newInstance( RAMFile );
+//		if (ramFile->open(file)) {
+//			file->close(); // is deleteonclose, so should delete.
+//			ramFile->deleteOnClose();
+//			return ramFile;
+//		}	else {
+//			ramFile->close();
+//			ramFile->deleteInstance();
+//		}
+//	}
+
+	return file;
+}
+
+void StdLocalFileSystem::update() 
+{
+}
+
+void StdLocalFileSystem::init() 
+{
+}
+
+void StdLocalFileSystem::reset() 
+{
+}
+
+//DECLARE_PERF_TIMER(StdLocalFileSystem_doesFileExist)
+Bool StdLocalFileSystem::doesFileExist(const Char *filename) const
+{
+	std::filesystem::path path = fixFilenameFromWindowsPath(filename, 0);
+	if(path.empty()) {
+		return FALSE;
+	}
+
+	std::error_code ec;
+	return std::filesystem::exists(path, ec);
+}
+
+void StdLocalFileSystem::getFileListInDirectory(const AsciiString& currentDirectory, const AsciiString& originalDirectory, const AsciiString& searchName, FilenameList & filenameList, Bool searchSubdirectories) const
+{
+
+	char search[_MAX_PATH];
+	AsciiString asciisearch;
+	asciisearch = originalDirectory;
+	asciisearch.concat(currentDirectory);
+	auto searchExt = std::filesystem::path(searchName.str()).extension();
+	if (asciisearch.isEmpty()) {
+		asciisearch = ".";
+	}
+
+	std::string fixedDirectory(asciisearch.str());
+
+#ifndef _WIN32
+	// Replace backslashes with forward slashes on unix
+	std::replace(fixedDirectory.begin(), fixedDirectory.end(), '\\', '/');
+#endif
+
+	strcpy(search, fixedDirectory.c_str());
+
+	Bool done = FALSE;
+	std::error_code ec;
+
+	auto iter = std::filesystem::directory_iterator(search, ec);
+	// The default iterator constructor creates an end iterator
+	done = iter == std::filesystem::directory_iterator();
+
+	if (ec) {
+		DEBUG_LOG(("StdLocalFileSystem::getFileListInDirectory - Error opening directory %s\n", search));
+		return;
+	}
+
+	while (!done)	{
+		if (!iter->is_directory() && iter->path().extension() == searchExt &&
+			(strcmp(iter->path().filename().string().c_str(), ".") && strcmp(iter->path().string().c_str(), ".."))) {
+			// if we haven't already, add this filename to the list.
+			// a stl set should only allow one copy of each filename
+			AsciiString newFilename = iter->path().string().c_str();
+			if (filenameList.find(newFilename) == filenameList.end()) {
+				filenameList.insert(newFilename);
+			}
+		}
+
+		iter++;
+		done = iter == std::filesystem::directory_iterator();
+	}
+
+	if (searchSubdirectories) {
+		auto iter = std::filesystem::directory_iterator(fixedDirectory, ec);
+
+		if (ec) {
+			DEBUG_LOG(("StdLocalFileSystem::getFileListInDirectory - Error opening subdirectory %s\n", fixedDirectory));
+			return;
+		}
+
+		// The default iterator constructor creates an end iterator
+		done = iter == std::filesystem::directory_iterator();
+
+		while (!done) {
+			if(iter->is_directory() && 
+				(strcmp(iter->path().filename().string().c_str(), ".") && strcmp(iter->path().string().c_str(), ".."))) {
```

Comment:
Can use a `const char* filenameStr = iter->path().filename().string().c_str()` to reduce long codes.

Reply:
When I made that suggestion I missed that it uses two different strings:

`(strcmp(iter->path().filename().string().c_str(), ".") && strcmp(iter->path().string().c_str(), "..")))`

path.filename and path. Sneaky. So this edit needs to be reverted.